src/vic3/data/commanders.rs

```diff
@@ -0,0 +1,123 @@
+use crate::block::Block;
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::desc::validate_desc;
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::modif::{validate_modifs, ModifKinds};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validator::Validator;
+
+#[derive(Clone, Debug)]
+pub struct CommanderOrder {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Vic3, Item::CommanderOrder, CommanderOrder::add)
+}
+
+impl CommanderOrder {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::CommanderOrder, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for CommanderOrder {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+
+        data.verify_exists(Item::Localization, key);
+        let loca = format!("{key}_desc");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("{key}_gerund");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+        let loca = format!("{key}_tooltip");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+
+        vd.field_item("texture", Item::File);
+        vd.field_choice("military_type", &["army", "navy"]);
+        vd.field_choice(
+            "ai_usage",
+            &["advance", "defend", "intercept", "raid_convoys", "escort_convoys"],
+        );
+        vd.field_bool("is_basic_order_type");
+
+        vd.field_validated_key_block("possible", |key, block, data| {
+            // TODO: this should have an `error_check` field somewhere inside
+            // TODO: `NOT = { has_trait = foo }` in this trigger will be misrepresented in the UI
+            let mut sc = ScopeContext::new(Scopes::Character, key);
+            validate_trigger(block, data, &mut sc, Tooltipped::No);
+        });
+        vd.field_validated_block("modifier", |block, data| {
+            let vd = Validator::new(block, data);
+            validate_modifs(block, data, ModifKinds::all(), vd);
+        });
+
+        // TODO: this should be a state inside a unit's Item::Entity
+        // The connection between the units and the entities is very indirect.
+        vd.field_value("entity_animation");
```

Comment:
Yes each unit's entity has a state with the given name, e.g. `state = { name = "defend" `.